### DISCUSSION TITLE
Authentication error in extensions

### DISCUSSION THREAD

**Discussion by Juanjo-GEx** [directus/directus#15492] [Archive]
### Describe the Bug

**There is an authentication error creating hooks.**

When adding a new record to a collection via Postman, Directus reports an access error. This error does not make much sense since the request has been made via bearer token with administration permissions.

### To Reproduce

- Make request via Rest API to a collection
    - The POST method is used to add a record to that collection.
    - Authentication is used through bearer token (_administration permissions, that is, everyone_)
- A hook type extension has been created to detect events, in this case it detects that a new record has been added.
- Directus detects the event, but when it tries to access the **updateOne method of services**, it shows the error.

[Video error](https://www.loom.com/share/34e0fe983fec4145bf0f0cd947c0ff46)

### Errors Shown

```console
An error was thrown while executing action "urls.items.create"
Error: You don't have permission to access this.
```

### What version of Directus are you using?

9.17.4

### What version of Node.js are you using?

16.16.0

### What database are you using?

SQLite 5.0.11

### What browser are you using?

Chrome

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
The permission error can be thrown by a variety of situations like non-existing collections, fields, items.
I have some questions as the video doesnt give a good look at the extension:
Are you using an `action` or `filter` to listen to the event?
Are you passing the `accountability` object when creating the itemsService?
Can you try adding `.catch((error) => console.error(error))` to the update and or read queries to hopefully catch some extra info on the error.
